tests/introspection/tests.py

```diff
@@ -170,6 +170,9 @@ def test_get_primary_key_column(self):
     def test_get_indexes(self):
         with connection.cursor() as cursor:
             indexes = connection.introspection.get_indexes(cursor, Article._meta.db_table)
+        if connection.vendor == 'postgresql':
```

Comment:
@timgraham Yes, that would be a nice functionality to add. But since adding that is a non-trivial task in itself, I guess I would prefer doing it when that functionality would have some actual use in Django, unless you think otherwise.

FWIW, from my past experience with working on https://github.com/django/django/pull/6982/commits/b94d5d3281566e872a68de975ca64d5c9e81fdb1, I know for one, that SQLite doesn't provide that functionality (but there are ways to get SQL query that had been used to create the index which can be hacked to get the index type). For two, I strongly believe Oracle would be having a sane way of doing this and implementing it would be relatively easy in Oracle. For MySQL, well, I've lost all hopes from MySQL, so I don't want to make any such claims for it.